Testing

VelzckC0D3 opened the pull request:
# Math Magicians -  `React` `ES6` `API`
## 👥 Pair programming with my partner: @Danfall369 

## ✨ _Requested Implementations_ ✨


✔️ _Set up React Testing Library following the instructions_
✔️ _Write unit test for `operate.js` & `calculate.js`_
✔️ _Created unit test for all the components_
✔️ _Used Jest snapshots to test the components_
✔️ _Used React Testing Library to simulate user interaction_
✔️ _All test passes_

#  💡 _Thank you for your review!_ 💡

VelzckC0D3 committed:
Testing

- added test components
- added test for the logic
- used Jest snapshots

Danfall369 committed:
Adding Test for Home.js and Calculator.js

both two are working.

VelzckC0D3 committed:
test & linters

- corrected all the linters and make all the test works

VelzckC0D3 committed:
Linters

corrected linter error

VelzckC0D3 committed:
Linters

Linters

OyePriscilla commented on src/components/Calculator.test.js:
```diff
@@ -0,0 +1,10 @@
+import { render } from '@testing-library/react';
+import Calculator from './Calculator';
+
+describe('Calculator', () => {
+  test('renders correctly', () => {
+    const { container } = render(<Calculator />);
+    expect(container.querySelector('.inputCont')).toBeTruthy();
+    expect(container.firstChild).toMatchSnapshot();
+  });
+});
```
- [x] Awesome implementing the snapshots for `Calcuator.js` and `Home.js`, can we please do the same for `Navbar.js` and `Qoutes.js`? You are to use jest snapshots to test all the components. 

<img width="443" alt="Untitled" src="https://user-images.githubusercontent.com/103386183/236598593-dbe17fc7-8a23-4c2f-96ef-7398a4db5fa5.png">

VelzckC0D3 replied:
Done! @OyePriscilla  thank you, I've already added a more professional and descriptive Readme, as well as the Snapshots to the logic testing components ! 🫡

OyePriscilla reviewed (changes requested):
Hi @VelzckC0D3 and @Danfall369 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![so-close](https://user-images.githubusercontent.com/103386183/196423863-0f522555-1b03-455e-a452-b822cdcfc398.gif)

## Highlights
Write unit tests with React Testing Library ✔️ 
Write unit tests with snapshots with Jest ✔️ 
Use of React Testing Library to simulate user interaction ✔️ 
All test passed ✔️ 

## Required Changes ♻️

- [ ] Kindly fix the broken links under the tech stack header of the README file, this will make your repo to be more engaging and professional. 👍 

## Optional suggestions

- [optional] Please I will advise you to change the text under the `tech stack` and `key feature` headers to something informative. 👍  

<img width="326" alt="Untitled" src="https://user-images.githubusercontent.com/103386183/236598829-3e671fa8-3218-4ad1-8561-d3b1b3e0776a.png">


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

VelzckC0D3 committed:
Update README.md

- As requested, added a personalized and professional Readme file for the project

VelzckC0D3 committed:
Jest Snapshot

- as requested, added jest snapshots to the Logic testing as well

VelzckC0D3 committed:
Merge branch 'testing' of https://github.com/VelzckC0D3/math_magicians_react into testing

levy002 reviewed (approved):
## Hello @VelzckC0D3 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Status: APPROVED :fire:  :fire: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
